jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/image/j9/ImageFactory.java

```diff
@@ -52,39 +53,53 @@
-	private static final String FACTORY_DDR = "com.ibm.j9ddr.view.dtfj.image.J9DDRImageFactory";
-	private final ArrayList<Exception> exceptions = new ArrayList<Exception>();
-	//log exceptions that occur when trying to find the correct image factory
+	private static final String FACTORY_DTFJ = "com.ibm.dtfj.image.j9.DTFJImageFactory"; //$NON-NLS-1$
+	private static final String FACTORY_DDR = "com.ibm.j9ddr.view.dtfj.image.J9DDRImageFactory"; //$NON-NLS-1$
+	private final ArrayList<Exception> exceptions = new ArrayList<>();
+	// log exceptions that occur when trying to find the correct image factory
 	private final Logger log = Logger.getLogger(DTFJ_LOGGER_NAME);
 	private ClassLoader imageFactoryClassLoader;
-	private File tmpdir = null;			//the directory which holds any extracted files
+	private File tmpdir = null; // the directory which holds any extracted files
 
 	static {
 		/* 
-		 * Following is to add exports java.base/jdk.internal.org.objectweb.asm 
-		 * programmatically to unnamed modules (j9ddr.jar) via reflection APIs.
+		 * Even though j9ddr.jar is a resource within the module openj9.dtfj,
+		 * the classes contained within appear at runtime to be in the un-named module.
+		 * The following programmatically exports jdk.internal.org.objectweb.asm
+		 * from java.base and certain packages from openj9.dtfj to the un-named
+		 * module via reflection APIs for use by j9ddr.jar.
 		 */
 		try {
-			Class<?> moduleClass = Class.forName("java.lang.reflect.Module");	
-			Class<?> modulesClass = Class.forName("jdk.internal.module.Modules");
-	 		Method	getModule = Class.class.getDeclaredMethod("getModule", new Class<?>[] {});
-
-			Method  addExportsToAllUnnamed = modulesClass.getDeclaredMethod("addExportsToAllUnnamed", new Class<?>[] {moduleClass, String.class});
-			addExportsToAllUnnamed.invoke(null, getModule.invoke(String.class), "jdk.internal.org.objectweb.asm");
-			
-			// following workaround is required after following com.ibm.dtfj packages are exported only to com.ibm.dtfjview
-			// these can be removed after j9ddr.jar is converted to a named module
-			addExportsToAllUnnamed.invoke(null, getModule.invoke(ImageFactory.class), "com.ibm.dtfj.image.j9");
-			addExportsToAllUnnamed.invoke(null, getModule.invoke(ImageFactory.class), "com.ibm.dtfj.utils.file");
-			addExportsToAllUnnamed.invoke(null, getModule.invoke(ImageFactory.class), "com.ibm.java.diagnostics.utils");
-			addExportsToAllUnnamed.invoke(null, getModule.invoke(ImageFactory.class), "com.ibm.java.diagnostics.utils.commands");
+			Class<?> modulesClass = Class.forName("jdk.internal.module.Modules"); //$NON-NLS-1$
+			Class<?> moduleClass;
+
+			try {
+				moduleClass = Class.forName("java.lang.Module"); //$NON-NLS-1$
+			} catch (ClassNotFoundException e) {
+				// fallback to name before b165
+				moduleClass = Class.forName("java.lang.reflect.Module"); //$NON-NLS-1$
+			}
+
+			Method exportToAll = modulesClass.getDeclaredMethod("addExportsToAllUnnamed", //$NON-NLS-1$ 
```

Comment:
I think the original variable name was more appropriate, why change it?

Reply:
I suggest the new name is just as descriptive and uses much less screen space making it easier to read.